packages/omgidl-parser/src/IDLNodeProcessor.ts

```diff
@@ -413,6 +415,14 @@ function fromScopedIdentifier(path: string): string[] {
   return path.split("::");
 }
 
+function getParentScopedIdentifier(scopedIdentifier: string): string | undefined {
+  const path = fromScopedIdentifier(scopedIdentifier);
+  if (path.length === 1) {
+    return undefined;
```

Comment:
You could return `"global scope"` here. This would let you change the return type to `string` rather than `string | undefined` and drop the `?? "global scope"` fallback above.

If this function were exported or some public api I'd probably frown on this but since it is local to this file and only called in one place you can make it serve that need.